frontend/src/metabase/palette/hooks/useCommandPalette.tsx

```diff
@@ -0,0 +1,187 @@
+import { useRegisterActions, type Action as KBarAction } from "kbar";
+import { useMemo } from "react";
+import { push } from "react-router-redux";
+import { t } from "ttag";
+
+import { getAdminPaths } from "metabase/admin/app/selectors";
+import { getSectionsWithPlugins } from "metabase/admin/settings/selectors";
+import {
+  useRecentItemListQuery,
+  useSearchListQuery,
+} from "metabase/common/hooks";
+import { getIcon, getName } from "metabase/entities/recent-items";
+import Search from "metabase/entities/search";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import * as Urls from "metabase/lib/urls";
+import { closeModal } from "metabase/redux/ui";
+import { getSettings } from "metabase/selectors/settings";
+import { Icon } from "metabase/ui";
+import type { SearchResult } from "metabase-types/api";
+
+export type PalettePageId = "root" | "admin_settings";
+
+export const useCommandPalette = ({
+  query,
+  debouncedSearchText,
+}: {
+  query: string;
+  debouncedSearchText: string;
+}) => {
+  const dispatch = useDispatch();
+
+  const hasQuery = query.length > 0;
+
+  const {
+    data: searchResults,
+    error: searchError,
+    isLoading: isSearchLoading,
+  } = useSearchListQuery<SearchResult>({
+    enabled: !!debouncedSearchText,
+    query: { q: debouncedSearchText, limit: 5 },
+    reload: true,
+  });
+
+  const { data: recentItems } = useRecentItemListQuery({
+    enabled: !debouncedSearchText,
+    reload: true,
+  });
+
+  const adminPaths = useSelector(getAdminPaths);
+  const settingValues = useSelector(getSettings);
+  const settingsSections = useMemo<Record<string, any>>(
+    () => getSectionsWithPlugins(),
+    [],
+  );
+
+  const docsAction = useMemo<KBarAction[]>(() => {
+    const ret: KBarAction[] = [
+      {
+        id: "search_docs",
+        name: query
+          ? `Search documentation for "${query}"`
+          : t`View documentation`,
+        section: "docs",
+        keywords: query, // Always match the query string
+        icon: () => <Icon name="document" />,
+        perform: () => {
+          const host = "https://www.metabase.com";
+          if (query) {
+            const params = new URLSearchParams({ query: query });
+            // TODO: find the documentation search URL in the right way
+            window.open(`${host}/search?${params}`);
```

Comment:
this seems right to me? or did you want to link it to a specific version's docs?